frontend/src/components/pages/CampCreation/ScheduleSessions/ScheduledSessionsCard.tsx

```diff
@@ -2,21 +2,54 @@ import React from "react";
 import { Box, HStack, Text, Button } from "@chakra-ui/react";
 import { CreateCampSession } from "../../../../types/CampsTypes";
 import SessionDayButton from "./SessionDayButton";
-import { getSessionDatesRangeString } from "../../../../utils/CampUtils";
+import {
+  getFormattedDateStringFromDateArray,
+  getSessionDates,
+} from "../../../../utils/CampUtils";
 
 type ScheduledSessionsCardProps = {
   currIndex: number;
   scheduledSession: CreateCampSession;
+  updateSession: (index: number, updatedSession: CreateCampSession) => void;
   onDelete: (index: number) => void;
 };
 
 const ScheduledSessionsCard = ({
   currIndex,
   scheduledSession,
+  updateSession,
   onDelete,
 }: ScheduledSessionsCardProps): JSX.Element => {
-  const weekDays: Map<string, boolean> = scheduledSession.selectedWeekDays;
-  const sessionDatesRangeString = getSessionDatesRangeString(scheduledSession);
+  const sessionDatesRangeString = getFormattedDateStringFromDateArray(
+    scheduledSession.dates,
+  );
+  const { selectedWeekDays } = scheduledSession;
+
+  const updateSelectedSessionDays = (day: string) => {
+    // set the new start date to the sunday in the same week
+    const updatedStartDate = new Date(scheduledSession.startDate.getTime());
+    updatedStartDate.setDate(
+      updatedStartDate.getDate() - scheduledSession.startDate.getDay(),
+    );
+
+    const updatedScheduledSession = scheduledSession;
+    const updatedWeekDays = updatedScheduledSession.selectedWeekDays;
+
+    const daySelected: boolean = updatedWeekDays.get(day) ?? false;
+    updatedWeekDays.set(day, !daySelected);
+
+    const updatedWeekDayValues = Array.from(updatedWeekDays.values());
+    const updatedDates: Date[] = getSessionDates(
+      updatedStartDate,
+      updatedWeekDayValues,
+    );
+
+    updatedScheduledSession.dates = updatedDates;
+    updatedScheduledSession.startDate = updatedStartDate;
+    updatedScheduledSession.endDate = updatedDates[updatedDates.length - 1];
```

Comment:
nit: could we create the updated object later? or does it not want you to modify the `scheduledSession` object?
```ts
... same processing on existing scheduledSession
    const updatedScheduledSession = {
      startDate: updatedStartDate,
      endDate: updatedDates[updatedDates.length - 1],
      dates: updatedDates,
      selectedWeekDays: updatedWeekDays,
    }
```